contrib/cdisetup/nvidia/nvidia.go

```diff
@@ -94,22 +96,15 @@ func (s *setup) Run(ctx context.Context) (err error) {
 	}
 
 	var needsDriver bool
-
-	if _, err := os.Stat("/proc/driver/nvidia"); err != nil {
-		needsDriver = true
-	}
-
-	var arch string
-	switch runtime.GOARCH {
-	case "amd64":
-		arch = "x86_64"
-	case "arm64":
-		arch = "sbsa"
-		// for non-sbsa could use https://nvidia.github.io/libnvidia-container/stable/deb
-	}
-
-	if arch == "" {
-		return errors.Errorf("unsupported architecture: %s", runtime.GOARCH)
+	if !hasDriversInstalled() {
+		// Check for nvidia-smi or /proc/driver/nvidia to determine if NVIDIA drivers are installed
+		if nvidiaSmi, err := exec.LookPath("nvidia-smi"); err == nil && nvidiaSmi != "" {
```

Comment:
this check only in combination with `isWSL()` ?

Reply:
No this is also valid on Linux with GPU request in container through `--gpus=all`:

```
  - containerPath: /usr/bin/nvidia-smi
    hostPath: /usr/bin/nvidia-smi
    options:
    - ro
    - nosuid
    - nodev
    - bind
```